### PR TITLE
$component controller and new router

### PR DESCRIPTION
**Pull request by samithaf**
Changes:

1. updated spec's to use $componentController. 
2. Updated ui-router version to latest which allows routes to use components instead templates

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ node_modules
 .settings
 *.log
 dist/
+.idea
```
Comment (fesor) [inline .gitignore]: Things like this should be handling via [global gitignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore).

**Comment by fesor** [comment]
Cool! The only thing that I don't like in `$componentController` is amount of boilerplate. Maybe it is possible to hide it somehow? 

p.s. Also not sure about uiRouter beta release... but I also use it and everything seems to be ok.

**Comment by samithaf** [comment]
@fesor In the very first look you feel like, there is lot of boilerplate goes on. However I believe it's all for good reason. Specially when you want to test a component with bindings and other dependancies like a service, it's very easy to unit test. Also I have added some additional unit tests for router as well. 

I started using uiRouter alpha release at work sometime back and I have a very good size team as well. Also recently upgraded to beta version and so far no issues. So I believe that beta release is good enough. 

The change in .gitignore I have revert it back.

Cheers.

**Comment by fesor** [comment]
@samithaf thanks again!

p.s. maybe we can hide boilerplate in some helper function but I'm not sure.

**Comment by pziegler** [comment]
Good work, but I think it would have been better to have completed this PR by updating ALL files..the generator template/hero files are still using the old style... looks a bit messy.
I"ll try to get this done today

**Comment by pziegler** [comment]
Added PR #165

**Pull request merged**